homeassistant/helpers/script.py

```diff
@@ -475,10 +476,11 @@ def async_script_wait(entity_id, from_s, to_s):
             async with async_timeout.timeout(timeout) as to_context:
                 await asyncio.wait(tasks, return_when=asyncio.FIRST_COMPLETED)
         except asyncio.TimeoutError as ex:
+            self._variables["wait"]["remaining"] = 0.0
             if not self._action.get(CONF_CONTINUE_ON_TIMEOUT, True):
                 self._log(_TIMEOUT_MSG)
+                trace_set_result(wait=self._variables["wait"], timeout=True)
```

Comment:
Since `wait` is already set, we just need timeout
```suggestion
                trace_set_result(timeout=True)
```

Reply:
https://github.com/home-assistant/core/pull/48100#discussion_r598661151